docs/architecture.md

```diff
@@ -0,0 +1,135 @@
+# Architecture
+
+This document provides an overview of the architecture and design principles of the NGINX Kubernetes Gateway. The target
+audience includes the following groups:
+
+* *Cluster Operators* who would like to know how the software works and also better understand how it can fail.
+* *Developers* who would like to [contribute][contribute] to the project.
+
+We assume that the reader is familiar with core Kubernetes concepts, such as Pod, Deployment, Service, and Endpoints.
+Additionally, we recommend reading [this blog post][blog] for an overview of the NGINX architecture.
```

Comment:
```suggestion
Additionally, we recommend reading [this blog post][blog] for an overview of the NGINX architecture.
```

I assume this might refer to an existing or upcoming blog post - flagging this to ensure it's updated.